homeassistant/components/sensor/__init__.py

```diff
@@ -546,6 +546,80 @@ class SensorStateClass(StrEnum):
+    SensorDeviceClass.DATA_SIZE: {
+        None,
+        SensorStateClass.MEASUREMENT,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.DATE: {None},
+    SensorDeviceClass.DISTANCE: {None}.union(set(SensorStateClass)),
+    SensorDeviceClass.DURATION: {None},
+    SensorDeviceClass.ENERGY: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.ENUM: {None},
+    SensorDeviceClass.FREQUENCY: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.GAS: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.HUMIDITY: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.ILLUMINANCE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.IRRADIANCE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.MOISTURE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.MONETARY: {None, SensorStateClass.TOTAL},
+    SensorDeviceClass.NITROGEN_DIOXIDE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.NITROGEN_MONOXIDE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.NITROUS_OXIDE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.OZONE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.PM1: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.PM10: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.PM25: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.POWER_FACTOR: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.POWER: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.PRECIPITATION: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.PRECIPITATION_INTENSITY: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.PRESSURE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.REACTIVE_POWER: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.SIGNAL_STRENGTH: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.SOUND_PRESSURE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.SPEED: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.SULPHUR_DIOXIDE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.TEMPERATURE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.TIMESTAMP: {None},
+    SensorDeviceClass.VOLATILE_ORGANIC_COMPOUNDS: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.VOLTAGE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.VOLUME: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.WATER: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.WEIGHT: {None, SensorStateClass.TOTAL},
```

Comment:
bthome/fitbit/tasmota/withings uses MEASUREMENT here.
TOTAL_INCREASING could also be used by the [Aussie Gold Hunters](https://en.wikipedia.org/wiki/Aussie_Gold_Hunters)
```suggestion
    SensorDeviceClass.WEIGHT: {None}.union(set(SensorStateClass)),
```

Reply:
> bthome/fitbit/tasmota/withings uses MEASUREMENT here.

Those need to be adjusted. This has been extensively discussed above a couple of time.

Total increasing isn't a viable thing at this point. None if the integrations can put it to use at this moment.